dockerspawner/dockerspawner.py

```diff
@@ -493,7 +493,7 @@ def get_ip_and_port(self):
             if resp is None:
                 raise RuntimeError("Failed to get port info for %s" % self.container_id)
             ip = resp[0]['HostIp']
-            port = resp[0]['HostPort']
+            port = int(resp[0]['HostPort'])
```

Comment:
Can you add a `log.warning` here showing the incoming value and type (repr should do)? I think there has to be a bug somewhere for this to be anything other than an int, so I'd like to at least make this not a silent conversion.